### PR TITLE
Update dynamo python for 2.13

### PR DESCRIPTION
**Pull request by pinzart90**
### Purpose

Dynamo (2.13) will no longer ship IronPython out of the box. It will only be available as a package on dynamopackages.com
This PR removes all direct dependencies on IronPython (and DSCPython3) and uses a new API to manage generic python Engines

### CODE REVIEW

**Comment by pinzart90** [comment]
FYI @mjkkirschner @QilongTang

**Review by mjkkirschner — APPROVED** [review]
LGTM - please add a small description to the PR template!

**Comment by pinzart90** [comment]
I could not get this solution to properly compile (keeps complaining about missing Revit Types ...even though I targeted Revit 2022 apis...)

File: src/DynamoRevit/Models/RevitDynamoModel.cs [hunk 1]
```diff
@@ -382,37 +381,36 @@ private void SetupPython()
...
+                engine.EvaluationStarted += (a, b, c) => ElementBinder.IsEnabled = false;
+                // register UnwrapElement method
+                engine.EvaluationStarted += (a, b, scopeSet) => scopeSet("UnwrapElement", unwrap);
+                // Turn on element binding after python script execution
+                engine.EvaluationFinished += (a, b, c, d) => ElementBinder.IsEnabled = true;
+            };
+            // Setup engines for all existing python engines
+            PythonServices.PythonEngineManager.Instance.AvailableEngines.ToList().ForEach(engine => setupPythonEngine(engine));
+            // Setup engines for any python engines that might be registered later on
+            PythonServices.PythonEngineManager.Instance.AvailableEngines.CollectionChanged += (object sender, NotifyCollectionChangedEventArgs e) =>
```
Comment (QilongTang) [inline src/DynamoRevit/Models/RevitDynamoModel.cs:405]: Do we need to unsubscribe these somewhere?

**Review by QilongTang — COMMENTED** [review]
One question then LGTM

File: src/DynamoRevit/Models/RevitDynamoModel.cs (hunk 1 above)
Comment (pinzart90) [reply src/DynamoRevit/Models/RevitDynamoModel.cs:405]: It did not unsubscribe them before...I did not want to change around too much
Will take a look if I can unbsub them easily

**Comment by ZiyunShang** [comment]
I am sorry that I missed this PR. I want to know that since Dynamo 2.13.0, Python Node does not support Python 2. Will the Python node in the old Dynamo file be automatically upgraded to CPython 3?

**Comment by ZiyunShang** [comment]
FYI: @ShengxiZhang @wangyangshi

**Review by ShengxiZhang — APPROVED** [review]

**Comment by pinzart90** [comment]
Tested and ready to merge
@mjkkirschner can you merge this ? I do not have permission

**Comment by mjkkirschner** [comment]
Just to expand a bit more on what @pinzart90  said - the reason not to change all engine types automatically from IronPython2 to CPython is that not all Python2 code will run in Cpython3 without changes - we have a migration tool that will help users perform this conversion node by node if they wish to use the new engine.

At the same time, if the users still *need* the old ironPython engine they can easily get it from the package manager using the workspace references dialog - it will display like any other missing package. that they can install directly from the workspace refs extension.

this blog post has more info:
https://dynamobim.org/dynamo-core-2-8-release/

**Comment by ZiyunShang** [comment]
Thanks for all your replies. LGTM, I will merge this.

**Review by ZiyunShang — APPROVED** [review]

**Comment by ZiyunShang** [comment]
Hi @pinzart , I just run the test file Revit API via Python.dyn with the latest Dynamo 2.13.0.3437. The "UnwrapElement" can't be recognized by Python node. Look at the picture. I remember I just test it before your last 2 commits here. May the new commits break the UnwrapElement registered.
CPython3Issue

**Comment by pinzart90** [comment]
> Hi @pinzart90 , I just run the test file Revit API via Python.dyn with the latest Dynamo 2.13.0.3437. The "UnwrapElement" can't be recognized by Python node. Look at the picture. I remember I just test it before your last 2 commits here. May the new commits break the UnwrapElement registered. CPython3Issue

I will take a look at it. It is possible

**Comment by ZiyunShang** [comment]
Well, another issue found - when I run the tests by RTF, run into `RunCurrentModel()` , all the python nodes run out with null.
SetupPython will be called here https://github.com/DynamoDS/DynamoRevit/blob/master/test/Libraries/RevitTestServices/RevitSystemTestBase.cs#L268 . But `SetupPythonEngine` and `OnPythonEngineCollectionChanged` are never triggered. Does it seem that Dynamo SystemTestBase can't support new Python Engine?
